test/puppetlabs/puppetdb/command_test.clj

```diff
@@ -1471,3 +1474,51 @@
                "Waited up to 5 seconds for 3 acknowledgement results")
 
            (is (= [nil nil nil] @ack-results))))))))
+
+(deftest missing-queue-files-accounted-for-in-stats
+  ;; Use a read-write lock to hold up command processing while we
+  ;; trash the queue.
+  (let [cmd-gate (Object.)
+        orig-handler message-handler]
+    (with-redefs [message-handler (fn [& wrap-args]
+                                    (let [h (apply orig-handler wrap-args)]
+                                      (fn [& handle-args]
+                                        (locking cmd-gate
+                                          (apply h handle-args)))))]
+      (svc-utils/with-puppetdb-instance
+        (let [dispatcher (get-service *server* :PuppetDBCommandDispatcher)
+              enqueue-command (partial enqueue-command dispatcher)
+              stats #(stats dispatcher)
+              deactivate (fn [name]
+                           (enqueue-command (command-names :deactivate-node)
+                                            3 name nil
+                                            (tqueue/coerce-to-stream
+                                             {:certname name
+                                              :producer_timestamp (Date.)})
+                                            ""))
+              depths #(mapv counters/value
+                            [(global-metric :depth)
+                             (cmd-metric "deactivate node" 3 :depth)])
+              qdir (get-path (conf/stockpile-dir (get-config)) "cmd" "q")
+              start-stats (stats)]
+          ;; Enqueue a few commands
+          (locking cmd-gate
```

Comment:
It appears that the whole body of the test is under this lock; does it actually work?